product_sticker/models/product_sticker.py

```diff
@@ -206,7 +206,12 @@ def _get_stickers(self, products, extra_domain=None):
         no_variant_attribute_lines = product_templates.attribute_line_ids.filtered(
             lambda al: al.attribute_id.create_variant == "no_variant"
         )
-        pp_pavs = products.product_template_variant_value_ids.product_attribute_value_id
+        pp_pavs = self.env["product.attribute.value"].browse()
+        for product in products:
+            if product.product_template_variant_value_ids:
+                pp_pavs |= product.product_template_variant_value_ids.product_attribute_value_id  # noqa: E501
+            else:
+                pp_pavs |= product.product_template_attribute_value_ids.attribute_line_id.value_ids  # noqa: E501
```

Comment:
suggestion: actually you can do all in one shot:


```suggestion
        products_with_variants = products.filtered("product_template_variant_value_ids")
        pp_pavs = products_with_variants.product_template_variant_value_ids.product_attribute_value_id | (products - products_with_variants).product_template_attribute_value_ids.attribute_line_id.value_ids
```

Reply:
I think is less confusing if I use an if statement. I've added a comment to describe the first part of the IF